pkg/genericapiserver/server_run_options.go

```diff
@@ -55,28 +64,104 @@ type ServerRunOptions struct {
-		RuntimeConfig:        make(config.ConfigurationMap),
-		SecurePort:           6443,
+		APIGroupPrefix:         "/apis",
+		APIPrefix:              "/api",
+		BindAddress:            net.ParseIP("0.0.0.0"),
+		CertDirectory:          "/var/run/kubernetes",
+		EnableLogsSupport:      true,
+		EnableProfiling:        true,
+		EnableWatchCache:       true,
+		InsecureBindAddress:    net.ParseIP("127.0.0.1"),
+		InsecurePort:           8080,
+		LongRunningRequestRE:   defaultLongRunningRequestRE,
+		MasterCount:            1,
+		MaxRequestsInFlight:    400,
+		MinRequestTimeout:      1800,
+		RuntimeConfig:          make(config.ConfigurationMap),
+		SecurePort:             6443,
+		StorageVersions:        registered.AllPreferredGroupVersions(),
+		DefaultStorageVersions: registered.AllPreferredGroupVersions(),
+	}
+}
+
+// StorageGroupsToEncodingVersion returns a map from group name to group version,
+// computed from the s.DeprecatedStorageVersion and s.StorageVersions flags.
+func (s *ServerRunOptions) StorageGroupsToEncodingVersion() (map[string]unversioned.GroupVersion, error) {
+	storageVersionMap := map[string]unversioned.GroupVersion{}
+	if s.DeprecatedStorageVersion != "" {
+		storageVersionMap[""] = unversioned.GroupVersion{Group: apiutil.GetGroup(s.DeprecatedStorageVersion), Version: apiutil.GetVersion(s.DeprecatedStorageVersion)}
+	}
+
+	// First, get the defaults.
+	if err := mergeGroupVersionIntoMap(s.DefaultStorageVersions, storageVersionMap); err != nil {
+		return nil, err
+	}
+	// Override any defaults with the user settings.
+	if err := mergeGroupVersionIntoMap(s.StorageVersions, storageVersionMap); err != nil {
+		return nil, err
+	}
+
+	return storageVersionMap, nil
+}
+
+// dest must be a map of group to groupVersion.
+func mergeGroupVersionIntoMap(gvList string, dest map[string]unversioned.GroupVersion) error {
+	for _, gvString := range strings.Split(gvList, ",") {
+		if gvString == "" {
+			continue
+		}
+		// We accept two formats. "group/version" OR
+		// "group=group/version". The latter is used when types
+		// move between groups.
+		if !strings.Contains(gvString, "=") {
+			gv, err := unversioned.ParseGroupVersion(gvString)
+			if err != nil {
+				return err
+			}
+			dest[gv.Group] = gv
+
+		} else {
+			parts := strings.SplitN(gvString, "=", 2)
+			gv, err := unversioned.ParseGroupVersion(parts[1])
+			if err != nil {
+				return err
+			}
+			dest[parts[0]] = gv
+		}
 	}
+
+	return nil
+}
+
+func (s *ServerRunOptions) NewClient() (clientset.Interface, error) {
+	clientConfig := &restclient.Config{
+		Host: net.JoinHostPort(s.InsecureBindAddress.String(), strconv.Itoa(s.InsecurePort)),
+		// Increase QPS limits. The client is currently passed to all admission plugins,
+		// and those can be throttled in case of higher load on apiserver - see #22340 and #22422
+		// for more details. Once #22422 is fixed, we may want to remove it.
+		QPS:   50,
+		Burst: 100,
+	}
+	if len(s.DeprecatedStorageVersion) != 0 {
```

Comment:
Great, I will send a PR to remove it (will let the flag remain, so that users dont get an error if they set it, but we will not use it anywhere in the code).